### ISSUE TITLE
Very strange interaction with rain

### ISSUE THREAD

**Issue by James-E-A** [nkomarn/harbor#75]
While myself and 2 friends were on a server, they tried to sleep to pass a thunderstorm during the night, and we saw the following spammed (neither the thunderstorm nor, I think, the night, were actually skipped -- the thunderstorm continued through most of my drafting this ticket):

![2020-12-28_20 57 40](https://user-images.githubusercontent.com/2776014/103257782-e3eaec00-494f-11eb-9d15-bb9422498916.png)

This occurred on 

* Harbor `v1.6.3`
* [PaperMC](https://papermc.io/) 1.16.4 `#348`
* Java 11 `OpenJDK 64-Bit Server VM 11.0.9+11`
* Fedora 31 `Linux 5.6.19-200.fc31.x86_64`

I had only changed 2 or 3 lines in my config:

```yml
# Harbor - Sleep mechanics enhancement plugin
# An open-source project by TechToolbox (@nkomarn)
# https://github.com/nkomarn/Harbor

# Important note regarding Essentials:
# Please make sure you negate the permission "essentials.sleepingignored" to
# prevent issues with Harbor and server operators when running Essentials.

night-skip:
  enabled: true # Skip the night if a percentage of the players in a world are sleeping
  percentage: 35 # Percentage of players required to be sleeping to skip the night (0 - 100)
  time-rate: 70 # The amount of ticks added to the current time every tick when skipping the night
  daytime-ticks: 1000 # The time in ticks that Harbor considers day
  instant-skip: true # Instantly skip the night when the percentage threshold is reached instead of showing the full animation
  proportional-acceleration: true # Increase the night skipping speed based on the amount of sleeping players
  clear-rain: true # Clear rain (if it's raining) when the night is skipped
  clear-thunder: true # Clear thunder (if it's thundering) when the night is skipped
  reset-phantom-statistic: true # Treats everyone online as if they have slept in the last 3 days after the night is skipped (check out /gamerule doInsomnia on 1.15+)

exclusions:
  ignored-permission: true # Exclude players with the permission "harbor.ignored" from the sleeping count
  exclude-adventure: true # Exclude players in adventure mode from the sleeping count
  exclude-creative: true # Exclude players in creative mode from the sleeping count
  exclude-spectator: true # Exclude players in spectator mode from the sleeping count
  exclude-vanished: true # Exclude vanished players from the sleeping count

# Detect AFK players and automatically remove them from the required sleeping count
# Essentials API is used for AFK detection when available- otherwise a fallback system is used
afk-detection:
  enabled: true
  timeout: 5 # Time in minutes until a player is considered AFK

# Blacklist for worlds- Harbor will ignore these worlds
blacklisted-worlds:
  - "world_nether"
  - "world_the_end"
whitelist-mode: false # Will treat the above list as a whitelist instead of a blacklist

messages:
  chat:
    enabled: true
    message-cooldown: 5 # The amount of time in seconds before a player can send another bed enter or leave message
    player-sleeping: "&e[player] is now sleeping ([sleeping]/[needed], [more] more needed to skip)."
    player-left-bed: "&e[player] got out of bed ([sleeping]/[needed], [more] more needed to skip)."
    night-skipping:
      - "&eAccelerating the night."
      - "&eRapidly approaching daytime."
    night-skipped:
      - "&eThe night has been skipped."
#      - "&eAhhh, finally morning."
#      - "&eArghh, it's so bright outside."
#      - "&eRise and shine."
  actionbar:
    enabled: true
    players-sleeping: "&e[sleeping] out of [needed] players are sleeping ([more] more needed to skip)"
    night-skipping: "&eEveryone is sleeping- sweet dreams!"
  bossbar:
    enabled: true
    players-sleeping:
      message: "&f&l[sleeping] out of [needed] are sleeping &7&l([more] more needed)"
      color: BLUE
    night-skipping:
      message: "&f&lEveryone is sleeping. Sweet dreams!"
      color: GREEN
  miscellaneous:
    chat-prefix: "&8&l(&6&lHarbor&8&l)&f "
    unrecognized-command: "Unrecognized command."

# Spooky internal controls
version: 1.6.3
interval: 1
metrics: true
debug: false
```

**Comment by nkomarn** [comment]
This is an issue with the instant skip option that should be fixed in the latest beta release (https://github.com/nkomarn/Harbor/releases/tag/1.6.4-b1). Give that a shot and see if it fixes your issue.

On another note, thanks for the detailed bug report; it's much appreciated.

**Comment by James-E-A** [comment]
This appears to have fixed it :+1:

> Haven't had any trouble with it  
> *–one of the players*

(Out of interest: _was_ this bug materially a duplicate of #72 — i.e. fixed by 81563404dbab9dcdcee9bbe58f2f71de4463eb9f — or was there a different commit that took care of this?)

**Comment by nkomarn** [comment]
Sort of; the previous fix didn't fix the bug when using the `instant-skip` option since I never bothered to test that on my server (I don't use the option).

Either way, glad to hear it's now fixed!
